src/UI/Component/Modal/Modal.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace ILIAS\UI\Component\Modal;
+
+use ILIAS\UI\Component\Component;
+use ILIAS\UI\Component\JavaScriptBindable;
+use ILIAS\UI\Component\Trigger\Triggerable;
+use ILIAS\UI\Component\Trigger\TriggerAction;
+
+/**
+ * This describes commonalities between the different modals
+ */
+interface Modal extends Component, JavaScriptBindable, Triggerable
+{
+
+    /**
+     * Get the title of the modal
+     *
+     * @return string
+     */
+    public function getTitle();
```

Comment:
Both could be possible I think. But we definitely need captions/descriptions per page.